npm1300 mfd: reset and hibernate

aasinclair opened the pull request:
Added reset and hibernate for npm1300

gmarull commented on drivers/reset/reset_npm1300.c:
```diff
@@ -0,0 +1,53 @@
+	const struct device *mfd;
+};
+
+static int reset_npm1300_toggle(const struct device *dev, uint32_t id)
```
`ARG_UNUSED(id)`?

I wonder if the reset API is well suited for this device (not sure what reset does in this case). The Reset API is oriented to IP blocks that control the reset lines of certain peripherals. See for example the DT-based API, where you'll typically have reset clients via `resets = <&resetctl ID>`. You can find example usages in ST/GD32 drivers.

aasinclair replied:
In this case it forces a full reset of the PMIC.  I agree it is tricky to decide where to put it.

gmarull replied:
maybe as an API in the mfd driver?

aasinclair replied:
How about adding reset & hibernate to the mfd driver (as they are fairly specific to the npm1300), and adding ship to regulators (as it is common to many devices)?

gmarull replied:
sounds good to me

aasinclair committed:
drivers: mfd: npm1300: Added reset and hibernate

New reset function which performs a full power reset

New hibernate function which powers down and wakes after
specified timeout

Signed-off-by: Andy Sinclair <andy.sinclair@nordicsemi.no>
